game/scripts/vscripts/Tutorial/SetupState.ts

```diff
@@ -0,0 +1,94 @@
+import { defaultSectionState, FilledSectionState, SectionState } from "./SectionState"
+import { getOrError, getPlayerHero } from "../util"
+
+function isObject<T>(item: T) {
+    return item && typeof item === "object" && !Array.isArray(item)
+}
+
+function mergeDeep(target: any, source: any): any {
```

Comment:
Seems dubious, we don't actually have any objects in the state right now, and even if we add some later are we sure we want to merge those instead of just overwriting? Seems like a potential source of unexpected behavior.

Reply:
Yea I wanted to have some more generic fields such as object descriptors which themselves have optional fields again. They would need their own default again too which seems not that easy to handle automatically and wouldn't work with this version of mergeDeep either.

Reply:
Removed this and put a note that says it wont work for nested if we need it later.